### PR TITLE
Add new URL /building-points-shp to get buildings as points

### PR DESCRIPTION
**Pull request by Gustry**
Add new URL /building-points-shp to get buildings as points

### CODE REVIEW

**Comment by Gustry** [comment]
I didn't put a QML file for QGIS 1 because I don't have any QGIS 1 on my computer :)
An error will occur while trying to copy the QML file. But is it really usefull to fix ? I hope everybody has at least 2.0.

File: reporter/osm.py [hunk 1]
```diff
@@ -230,7 +230,76 @@ def extract_buildings_shapefile(
         qgis_version)
 
 
-def extract_roads_shapefile(file_path, qgis_version=1, output_prefix=''):
+def extract_buildings_points_shapefile(file_path, qgis_version=2, output_prefix=''):
```
Comment (timlinux) [inline reporter/osm.py]: extract_building_points_shapefile (building should be singular)

File: reporter/osm.py [hunk 2]
```diff
@@ -230,7 +230,76 @@ def extract_buildings_shapefile(
...
+    """Convert the OSM xml file to a buildings points shapefile.
+
+    This is a multistep process:
+        * Create a temporary postgis database
+        * Load the osm dataset into POSTGIS with osm2pgsql and our custom
+             style file.
+        * Save the data out again to a shapefile
```
Comment (timlinux) [inline reporter/osm.py]: Above this line you should add :
* Calculate point on surface for building centroids

File: reporter/osm.py [hunk 3]
```diff
@@ -230,7 +230,76 @@ def extract_buildings_shapefile(
...
+    :param file_path: Path to the OSM file name.
+    :type file_path: str
+
+    :param qgis_version: Get the QGIS version. Currently 1,
+        2 are accepted, default to 2. A different qml style file will be
+        returned depending on the version
+    :type qgis_version: int
+
+    :param output_prefix: Base name for the shape file. Defaults to ''
+        which will result in an output file of 'buildings.shp'. Adding a
```
Comment (timlinux) [inline reporter/osm.py]: 'building_points.shp'

File: reporter/osm.py [hunk 4]
```diff
@@ -230,7 +230,76 @@ def extract_buildings_shapefile(
...
+        prefix of e.g. 'test-' would result in a downloaded file name of
+        'test-buildings.shp'. Allowed characters are [a-zA-Z-_0-9].
```
Comment (timlinux) [inline reporter/osm.py]: test-building_points.shp

File: reporter/osm.py [hunk 5]
```diff
@@ -230,7 +230,76 @@ def extract_buildings_shapefile(
...
+    :returns: Path to zipfile that was created.
+    :rtype: str
+
+    """
+    if not check_string(output_prefix):
+        error = 'Invalid output prefix: %s' % output_prefix
+        LOGGER.exception(error)
+        raise Exception(error)
+
+    feature_type = 'buildings-points'
```
Comment (timlinux) [inline reporter/osm.py]: building_points

File: reporter/resources/buildings-points/buildings-points.keywords [hunk 6]
```diff
@@ -0,0 +1,5 @@
+category: exposure
+subcategory: structure
+title: Buildings
```
Comment (timlinux) [inline reporter/resources/buildings-points/buildings-points.keywords]: title: Building centroids

File: reporter/resources/buildings-points/transform.sql [hunk 7]
```diff
@@ -0,0 +1,78 @@
+ALTER TABLE planet_osm_polygon ADD COLUMN "type" VARCHAR(255) NULL;
```
Comment (timlinux) [inline reporter/resources/buildings-points/transform.sql]: This file violates DRY (dont repeat yourself). Can you find a way that it and others are included once only?

File: reporter/test/test_osm.py [hunk 8]
```diff
@@ -61,6 +62,12 @@ def test_extract_buildings_shapefile(self):
         #print zip_path
         self.assertTrue(os.path.exists(zip_path), zip_path)
 
+    def test_extract_buildings_points_shapefile(self):
```
Comment (timlinux) [inline reporter/test/test_osm.py]: building singular

File: reporter/test/test_osm.py [hunk 9]
```diff
@@ -61,6 +62,12 @@ def test_extract_buildings_shapefile(self):
...
+        """Test the osm to shp converter."""
+        zip_path = extract_buildings_points_shapefile(FIXTURE_PATH)
```
Comment (timlinux) [inline reporter/test/test_osm.py]: building

File: reporter/views.py [hunk 10]
```diff
@@ -20,6 +20,7 @@
 from .osm import (
     get_osm_file,
     extract_buildings_shapefile,
```
Comment (timlinux) [inline reporter/views.py]: building (singular)

**Comment by timlinux** [comment]
Can you reference building_points throughout instead of buildings_points and I had a few other minor comments above. Once all are addressed, LGTM to merge. Thanks!

File: reporter/resources/buildings-points/transform.sql (hunk 7 above)
Comment (Gustry) [inline reporter/resources/buildings-points/transform.sql]: Only transform.sql ?
There are a lot of files like this one : default.style, licence, ...

But transform.sql should disapear with https://github.com/AIFDR/inasafe/issues/1664
Should I fix this ?

**Pull request merged**